extensions/BugModal/lib/WebService.pm

```diff
@@ -62,6 +81,25 @@ sub rest_resources {
     ]
 }
 
+sub products {
+    my $user = Bugzilla->user;
+    my @products = @{ $user->get_enterable_products };
+
+    @products = map { { name => $_->name } } @products;
+    return {
+        products     => \@products,
+    };
+}
+
+sub components {
+    my ($self, $params) = @_;
+    my $product = Bugzilla::Product->check({ name => $params->{product_name}, cache => 1 });
+    my $components = _name($product->components);
+    my %options;
+    $options{component} = $components;
+    return \%options;
```

Comment:
Yes, it does return in the format you specified and is same as that of `products`.